### ISSUE TITLE
Adding a number to a vector changes the vector itself

### ISSUE THREAD

**Issue by meijeru** [red/red#2216]
See this fragment:
```
v: make vector! [1.0 2.0 3.0]
v + 1.0 ; => make vector! [2.0 3.0 4.0]
v ; => make vector! [2.0 3.0 4.0]
```
The same happens with integers. I am sure this is not what is intended.

**Comment by meijeru** [comment]
Apparently (see gitter/red) the semantics are still to be defined. I can see two analogies:
with tuples:
```v: 1.2.3
v + 1 ; => 2.3.4 and v is unchanged
```

**Comment by greggirwin** [comment]
If vector args are changed in place, the doc string needs to say so. As long as we know how to think of them, and how they behave, as with other types, either design choice is valid. I have `tuple!` funcs just to do mutation, incrementing build numbers, versions, etc.

**Comment by Oldes** [comment]
If I can vote, I think current behavior is correct.

**Comment by meijeru** [comment]
For consistency's sake, those who vote for the current behaviour of vector! + number! should also want the vector! + vector!  case to be mutable, i.e. changing the first vector. @Oldes do you agree?

**Comment by Oldes** [comment]
Yes... I think so... you can always use ```copy``` when you need a new value. But it would probably require to see, what scenario would be more common.. I guess that ```matrix!``` type would be just ```vector!``` internally, so one could look at some matrix math scenarios (i'm not good in math;)

**Comment by meijeru** [comment]
In any case, the current implementation is not consistent either way...

**Comment by Oldes** [comment]
One should first answer a question, _what is main purpose of ```vector!``` datatype?_ If it is speed and memory efficiency, than creating a copy on each operation is a nonsense.

**Comment by Oldes** [comment]
I'm using Starling framework elsewhere, so I can compare its [matrix utils](https://github.com/Gamua/Starling-Framework/blob/master/starling/src/starling/utils/MatrixUtil.as ): You can see that all main functions (_skew, prependMatrix, prependTranslation, prependScale, prependRotation, prependSkew_) modifies existing matrix input and other functions, like _convertTo3D_, accepts matrix as an input argument to avoid allocations.

**Comment by Oldes** [comment]
As I'm now using vectors to hold thousands values, I'm pretty sure that current behaviour is the good one. I definitely don't want to have copy of such a big arrays in memory on each math operation.

**Comment by hiiamboris** [comment]
It's absurd from the design point of view, totally inconsistent and totally unexpected. I support @meijeru 100%.
```
>> a: make vector! [1 1 1 1 1]
== make vector! [1 1 1 1 1]
>> b: make vector! [1 1 1 1 1]
== make vector! [1 1 1 1 1]
>> a + b
== make vector! [2 2 2 2 2]
>> a
== make vector! [1 1 1 1 1]
>> b
== make vector! [1 1 1 1 1]
>> a + 1
== make vector! [2 2 2 2 2]
>> a
== make vector! [2 2 2 2 2]
```
I however also agree with @Oldes that a modifying function is a must, but it should be separate from the `+` operator. Something in the lines of `math`, that will take an expression (e.g. `[a + b + 2 * 3]`) and modify the first vector inside that expression in place.

**Comment by hiiamboris** [comment]
See also my thoughts here https://github.com/red/REP/issues/10#issuecomment-660499053
